data_analysis/dataframes.py

```diff
@@ -0,0 +1,105 @@
+"""
+This script contains methods for creating pandas DataFrames from relations of a database using sql-statements
+"""
+
+import pandas as pd
+import sqlalchemy
+from loguru import logger
+
+import dql_scripts.select_joins
+from db_engines.db_sources_data.sql_server_test_localhost import SQLServerTestDBs
+from db_engines.sql_server_engine import url_SQLServerTestDBMS
+from phillip.db_connection import create_session_from_url
+from dql_scripts import select_joins as jnt_sel
+from db_engines.sql_server_engine import engine_sqlservertest_main as sqlserver_engine
+
+
+def create_df_from_list(rows_data_list: list, columns_list: list) -> pd.DataFrame:
+    """
+    Creates a pandas DataFrame based on a list of selection result Rows instances that were already created using f.e.
+    SQL Alchemy Core API or ORM API
+    :param rows_data_list: a list of Rows instances of a select result
+    :param columns_list: an iterable with the strings of attributes names of the select result
+    :return: a pandas dataframe with the data of the list with the selection result
+    """
+    result_df = pd.DataFrame(rows_data_list, columns=columns_list)
+    logger.debug("A DataFrame with following parameters was created from the list: \n", result_df.info())
+
+    return result_df
```

Comment:
Für die Umwandlung einer Liste in einen DF extra eine Funktion ? ;)

Reply:
Tja danke für den Hinweis 😆 Mir ist während der Verbesserung des Codes irgendwie komplett entlaufen, dass ich hier was anderes machen wollte... Meine Idee war eigentlich noch eine statische Typ-Prüfung einzuführen. Ansonsten macht die Methode gar keinen Sinn.

Ich habe sie jetzt mit Union Type Annotations erweitert, damit nur Rows von SQL Alchemy akzeptiert werden. [Hier ist der Commit ](https://github.com/il-ia-ni/python_sql_tutorial/pull/4/commits/7619049e3e5a30808c6adff4f55b56fa48b0f7bb) dazu. Auch den Methodennamen habe ich angepasst. 

Theoretisch könnte man weitere Validierungen von Rows vor der Erstellung von DataFrames einführen. Es war auf jeden Fall eine gute Übung zu den Type Annotations in Python 😉